mojo/core/node_controller.cc

```diff
@@ -516,6 +533,41 @@ void NodeController::SendBrokerClientInvitationOnIOThread(
   channel->AcceptInvitee(name_, temporary_node_name);
 }
 
+void NodeController::RetryInvitationOnIOThread(
+    ScopedProcessHandle old_process,
+    ScopedProcessHandle target_process,
+    ConnectionParams connection_params) {
+  auto broker_it = broker_hosts_.find(old_process.get());
+  broker_it->second->ResetBrokerChannel(std::move(connection_params));
+
+  PlatformChannel node_channel;
+  ConnectionParams node_connection_params(node_channel.TakeLocalEndpoint());
+  bool channel_ok = broker_it->second->SendChannel(
+      node_channel.TakeRemoteEndpoint().TakePlatformHandle());
+
+  #if defined(OS_WIN)
+  if (!channel_ok) {
+    // On Windows the above operation may fail if the channel is crossing a
+    // session boundary. In that case we fall back to a named pipe.
+    NamedPlatformChannel::Options options;
+    NamedPlatformChannel named_channel(options);
+    node_connection_params =
+        ConnectionParams(named_channel.TakeServerEndpoint());
+    broker_host->SendNamedChannel(named_channel.GetServerName());
+  }
+#else
+  CHECK(channel_ok);
+#endif  // defined(OS_WIN)
+
+  auto it = pending_invitations_.find(temp_node_name_);
```

Comment:
BrokerCastanets already has a node_channel pointer. See broker_castanets.h:126

Reply:
Done. I added a getter for it.